src/inmanta/logging.py

```diff
@@ -423,7 +690,107 @@ def format(self, record: logging.LogRecord) -> str:
+            if match:
+                # Log record created by an Inmanta module.
+                return match.groupdict()["module_name"]
+            else:
+                # Log record created by Inmanta code.
+                return logger_mode.value
+        else:
+            # Don't modify the logger name
+            return logger_name
+
+
+class MultiFileHandler(logging.Handler):
+    """
+    A file handler that writes to different files based on the last part of the logger name.
+    """
+
+    def __init__(self, name_parent_logger: str, log_file_template: str) -> None:
+        """
+        :param name_parent_logger: The log records created by children of this logger will be written to file by this handler.
+                                   This handler will ignore log records created by any other logger.
+        :param log_file_template: A template for the path to the log file. This is an f-string that holds the parameter
+                                  `child_logger_name`. This part will be replaced by the (non-fully-qualified) name of
+                                  the child logger self.name_parent_logger.
+        """
+        super().__init__()
+        self.name_parent_logger = name_parent_logger
+        self.log_file_template: str = log_file_template
+        self.child_handlers: dict[str, handlers.WatchedFileHandler] = {}
+
+    def flush(self) -> None:
+        for child in self.child_handlers.values():
+            child.flush()
+
+    def setFormatter(self, fmt: Optional[logging.Formatter]) -> None:
+        super().setFormatter(fmt)
+        for child in self.child_handlers.values():
+            child.setFormatter(fmt)
+
+    def setLevel(self, level: str | int) -> None:
+        super().setLevel(level)
+        for child in self.child_handlers.values():
+            child.setLevel(level)
+
+    def close(self) -> None:
+        for child in self.child_handlers.values():
+            child.close()
+        self.child_handlers = {}
+        super().close()
+
+    def emit(self, record: logging.LogRecord) -> None:
+        if not record.name.startswith(f"{self.name_parent_logger}."):
+            return
+        path_logfile = self.log_file_template.format(child_logger_name=record.name.removeprefix(f"{self.name_parent_logger}."))
+        if path_logfile not in self.child_handlers:
+            try:
+                handler = logging.handlers.WatchedFileHandler(filename=path_logfile, mode="a+")
+            except FileNotFoundError:
+                log_dir = os.path.dirname(self.log_file_template)
+                if not os.path.exists(log_dir):
+                    LOGGER.warning(
+                        "Cannot write to resource action log, because directory %s doesn't exist.",
+                        path_logfile,
+                    )
+                else:
+                    LOGGER.exception("Failed to write to resource action log file.")
```

Comment:
Perhaps include the exception in this case? Or does `logging.exception` already do that?

Reply:
Yes. This method automatically adds the exception information to the log message.